test/validation/api/time/time.c

```diff
@@ -543,54 +564,16 @@ static void time_test_accuracy_nsec(time_nsec_cb time_nsec)
 		tsdiff.tv_nsec = ts2.tv_nsec - ts1.tv_nsec;
 		tsdiff.tv_sec = ts2.tv_sec - ts1.tv_sec;
 	}
-
-	diff  = t2 - t1;
-	sec_t = ((double)diff) / ODP_TIME_SEC_IN_NS;
 	sec_c = ((double)(tsdiff.tv_nsec) / 1000000000L) + tsdiff.tv_sec;
 
-	/* Check that ODP time is within +-5% of system time */
-	CU_ASSERT(sec_t < sec_c * 1.05);
-	CU_ASSERT(sec_t > sec_c * 0.95);
-}
-
-static void time_test_local_accuracy(void)
-{
-	time_test_accuracy(odp_time_local, odp_time_local_from_ns);
-}
-
-static void time_test_global_accuracy(void)
-{
-	time_test_accuracy(odp_time_global, odp_time_global_from_ns);
-}
-
-static void time_test_local_strict_accuracy(void)
-{
-	time_test_accuracy(odp_time_local_strict, odp_time_local_from_ns);
-}
-
-static void time_test_global_strict_accuracy(void)
-{
-	time_test_accuracy(odp_time_global_strict, odp_time_global_from_ns);
-}
-
-static void time_test_local_accuracy_nsec(void)
-{
-	time_test_accuracy_nsec(odp_time_local_ns);
-}
-
-static void time_test_global_accuracy_nsec(void)
-{
-	time_test_accuracy_nsec(odp_time_global_ns);
-}
+	for (i = 0; i < 4; i++) {
+		uint64_t diff  = t2[i] - t1[i];
+		double sec_t = ((double)diff) / ODP_TIME_SEC_IN_NS;
 
-static void time_test_local_strict_accuracy_nsec(void)
-{
-	time_test_accuracy_nsec(odp_time_local_strict_ns);
-}
-
-static void time_test_global_strict_accuracy_nsec(void)
-{
-	time_test_accuracy_nsec(odp_time_global_strict_ns);
+		/* Check that ODP time is within +-5% of system time */
+		CU_ASSERT(sec_t < sec_c * 1.05);
+		CU_ASSERT(sec_t > sec_c * 0.95);
+	}
```

Comment:
When one test fails with a loop, it's hard to tell from a log which one failed. Maybe unroll the loop, or print out something on a failure. Also the loop on line 520 could be changed the same way.